feg/gateway/docker/python/Dockerfile

```diff
@@ -1,58 +1,69 @@
-# -----------------------------------------------------------------------------
-# Builder image to generate proto files
-# -----------------------------------------------------------------------------
+################################################################################
+# Copyright 2023 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
 FROM ubuntu:focal AS builder
 
+ARG DEB_PORT=amd64
 # workaround to avoid interactive tzdata configuration
 ARG DEBIAN_FRONTEND=noninteractive
 
-# Install the runtime deps from apt.
-RUN apt-get -y update && apt-get -y install curl make virtualenv zip \
-    apt-utils software-properties-common apt-transport-https git openjdk-8-jdk ant
-
-# Install protobuf compiler.
-RUN curl -Lfs https://github.com/protocolbuffers/protobuf/releases/download/v3.1.0/protoc-3.1.0-linux-x86_64.zip -o protoc3.zip && \
-    unzip protoc3.zip -d protoc3 && \
-    mv protoc3/bin/protoc /usr/bin/protoc && \
-    chmod a+rx /usr/bin/protoc && \
-    cp -r protoc3/include/google /usr/include/ && \
-    chmod -R a+Xr /usr/include/google && \
-    rm -rf protoc3.zip protoc3
-
-RUN apt-get -y update && apt-get -y install python3.8
+# Install the build deps with apt.
+RUN apt-get -y update && apt-get -y install \
+    apt-transport-https \
+    apt-utils \
+    git \
+    libsystemd-dev \
+    python3.8 \
+    python3-distutils \
+    python3-pip \
+    pkg-config \
+    systemd \
+    wget
 
 ENV MAGMA_ROOT /magma
-ENV PYTHON_BUILD /build/python
 ENV PIP_CACHE_HOME ~/.pipcache
-ENV CODEGEN_ROOT /var/tmp/codegen
-ENV CODEGEN_VERSION 2.2.3
-ENV SWAGGER_CODEGEN_DIR $CODEGEN_ROOT/modules/swagger-codegen-cli/target
-ENV SWAGGER_CODEGEN_JAR $SWAGGER_CODEGEN_DIR/swagger-codegen-cli.jar
 ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
-RUN printenv > /etc/environment
 
-# Download swagger codegen
-RUN mkdir -p ${SWAGGER_CODEGEN_DIR}; \
-    curl -Lfs https://repo1.maven.org/maven2/io/swagger/swagger-codegen-cli/${CODEGEN_VERSION}/swagger-codegen-cli-${CODEGEN_VERSION}.jar -o ${SWAGGER_CODEGEN_JAR}
+# Download Bazel
+RUN  wget -P /usr/sbin --progress=dot:giga https://github.com/bazelbuild/bazelisk/releases/download/v1.10.0/bazelisk-linux-"${DEB_PORT}" && \
```

Comment:
Do we have a way to define the `bazelisk` version globally? I found about 10 instances of the version being hardcoded in different Dockerfiles across the project.

Reply:
It would be possible to refactor this into a wrapper script, but I think it is not worth the added complexity at the moment. Bazelisk is just an installer for bazel and the version is not that important. The Bazel version is set by the `.bazelversion` file.

Reply:
I created an issue https://github.com/magma/magma/issues/15114, but the priority for this is low imo.

Reply:
Sounds good and I agree this being low prio